shim_maker.py

```diff
@@ -21,62 +26,138 @@ def fancy_list(prefix: str, to_join: Iterable[str], limit: int = 100, with_brack
+        ):
+            members.append(memb)
+    code = fancy_list(f"from {module_name} import ", members)
 
     try:
-        public_members = [f"\"{memb}\"" for memb in subpckg.__all__]
+        public_members = [f'"{memb}"' for memb in subpckg.__all__]
     except AttributeError:
         public_members = None
 
     if public_members:
         new_all = fancy_list("__all__ = ", public_members, with_brackets=True)
-        imports += "\n\n" + new_all
-
-    imports += f"\n\nfrom {module_name} import __dict__ as __original_dict__\nlocals().update(__original_dict__)"
-
-    return imports + "\n"
-
-
-def shim_folder(path: str, pypath: str, shim_path: str) -> None:
-    for fn in os.listdir(path):
-        if fn in ("__init__.py", "__main__.py", "py.typed"):
-            with open(f"{path}/{fn}", "r", encoding="utf-8") as f:
-                content = f.read()
-            with open(f"{shim_path}/{fn}", "w", encoding="utf-8") as f:
-                f.write(content)
-
-        elif fn.startswith("__"):
-            pass
+        code += "\n\n" + new_all
+
+    code += f"\n\n# isort: split\nfrom {module_name} import __dict__ as __original_dict__\nlocals().update(__original_dict__)"
+
+    return docstring, sort_imports(code)
+
+
+def shim_folder(path: pathlib.Path, pypath: str, shim_path: pathlib.Path) -> List[pathlib.Path]:
+    modified = []
+    for fn in path.iterdir():
+        mod = fn.relative_to(path)
+
+        if not (shim_path / mod).parent.exists():
+            (shim_path / mod).parent.mkdir(exist_ok=True, parents=True)
+            print("Created directory: ", (shim_path / mod))
+
+        if fn.name in ("__init__.py", "__main__.py", "py.typed") and "tasks" != fn.parent.name:
+            with open(path / mod, "r", encoding="utf-8") as fr:
+                try:
+                    with open(shim_path / mod, "r", encoding="utf-8") as fw:
+                        if fw.read() == fr.read():
+                            continue
+                except FileNotFoundError:
+                    pass
+                fr.seek(0)
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as fw:
+                    fw.write(fr.read())
+            modified.append(shim_path / mod)
+
+        elif fn.suffix == ".py":
+            docstring, imports = create_file(f"{pypath}.{mod.stem}")
+            to_write = ""
+            if docstring:
+                to_write += '"""\n' + docstring.strip() + '\n"""\n\n'
+            to_write += imports
+            try:
+                with open(shim_path / mod) as f:
+                    existing = f.read()
+            except FileNotFoundError:
+                existing = None
+            if existing != to_write:
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as f:
+                    f.write(to_write)
+
+                modified.append(shim_path / mod)
```

Comment:
could also be simplified to
```suggestion

            if shim_mod.is_file():
                existing = shim_mod.read_text(encoding="utf-8")
            else:
                existing = None
            if existing != to_write:
                print("Updating file: ", shim_mod)
                shim_mod.write_text(to_write, encoding="utf-8")
                modified.append(shim_mod)
```